icu4j/main/tests/core/src/com/ibm/icu/dev/test/calendar/JapaneseTest.java

```diff
@@ -450,5 +450,43 @@ public void TestLimits() {
         doLimitsTest(jcal, null, cal.getTime());
         doTheoreticalLimitsTest(jcal, true);
     }
+
+    // The following currently assumes that Reiwa is the last known/valid era.
+    // Filed ICU-20551 to generalize this when we have more time...
+    @Test
+    public void TestJpnCalAddSetNextEra() {
+        final JapaneseCalendar jCal = new JapaneseCalendar();
+        jCal.clear();   // This sets to 1970 in Showa
+
+        final int sEra = jCal.get(Calendar.ERA);  // Don't assume era number for Showa
+        final int[] startYears = { 1926, 1989, 2019, 0 };    // start years for Show, Heisei, Reiwa; 0 for invalid era beyond
+
+        for (int iEra = 1; iEra < 3; iEra++) {
+            jCal.clear();
+            jCal.set(Calendar.ERA, sEra + iEra);
+            int eYear = jCal.get(Calendar.EXTENDED_YEAR);
+            if (startYears[iEra] == 0) {
```

Comment:
We will never hit this block with loop limit iEra < 3 and it tests for behavior that we decided not to support, so we should just eliminate the block as in the ICU4C implementation

Reply:
Thanks. Yes, I forgot to remove this.. Updated now.